lib/src/model/result.dart

```diff
@@ -53,11 +52,12 @@ class Result<Value, Failure> {
       ((17 * 37) + (_value?.hashCode ?? 0)) * 37 + (_failure?.hashCode ?? 0);
 
   @override
-  bool operator ==(dynamic other) =>
+  bool operator ==(Object other) =>
+      other is Result &&
       runtimeType == other.runtimeType &&
       _value == other._value &&
       _failure == other._failure;
 
-  final Value _value;
-  final Failure _failure;
+  final Value? _value;
```

Comment:
Should not [be forced to] be optional [anymore]. If needed, the class can be specialized with an optional type (`Result<int?, String>`).

Reply:
I am not 100% about this. If we create a failure we get: 

```dart
const Result.failure(this._failure)
      : assert(_failure != null),
        _value = null; 
```
how do you want to handle this case?

Reply:
Huh, good point 🤦‍♂️ Okay, I see now, why this is needed.

There are other ways to model sum types [that would allow us to avoid optionality here], but let's just stick with this.